lib/prima_auth0_ex/absinthe/require_permissions.ex

```diff
@@ -0,0 +1,39 @@
+defmodule PrimaAuth0Ex.Absinthe.RequirePermissions do
+  @moduledoc """
+  Absinthe middleware that ensure the permission is included in the current security context.
+  """
+
+  require Logger
+
+  alias PrimaAuth0Ex.Absinthe.CreateSecurityContext.Context
+
+  @behaviour Absinthe.Middleware
+
+  @impl true
+  def call(
+        %{context: %Context{permissions: permissions}} = resolution,
+        required_permissions
+      ) do
+    if has_required_permissions?(permissions, required_permissions) do
+      resolution
+    else
+      resolve(resolution, required_permissions)
+    end
+  end
+
+  defp has_required_permissions?(nil = _permissions, _required_permissions), do: false
+
+  defp has_required_permissions?(permissions, required_permissions),
+    do: Enum.all?(required_permissions, fn required_permission -> required_permission in permissions end)
```

Comment:
```suggestion
    do: Enum.all?(required_permissions, &Enum.member?(required_permissions, &1))
```
Dunno if I like it more this way :laughing: